graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1571,6 +1572,18 @@ def write_edge_structure(
             how="inner",
         )
 
+        # Always create the order column for the edges dataframe before writing
+        original_schema = edge_df_with_int_ids.schema
+        edge_df_with_int_ids = edge_df_with_int_ids.rdd.zipWithIndex()
+
+        new_schema = original_schema.add(StructField(EDGE_MAPPING_INT, LongType(), False))
+
+        edge_rdd_with_order_ids = edge_df_with_int_ids.map(
+            lambda rdd_row: (list(rdd_row[0]) + [rdd_row[1]])  # type: ignore
+        )
+
+        edge_df_with_int_ids = self.spark.createDataFrame(edge_rdd_with_order_ids, new_schema)
+
```

Comment:
Have we tested just using e.g. the `dst` column for order? That's what DGL does IIRC

Reply:
I can try it.

Reply:
Test with only the `dst` column, in this way the order of the whole dataframe will be undetermined. Let us move with the int id for now.

Reply:
What do we mean undetermined? What's happening when sorting by `dst` vs. creating a new column?

Reply:
I mean instead of using newly created `EDGE_MAPPING_INT` column, I use `dst` here for the order_col. When doing the `transformed_label = transformed_label.sort(self.order_col).cache()` to maintain the order, then the final output order will be totally undetermined. So I suggest we keep using a new column.